x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/store/middleware.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FoundExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import { CoreStart, HttpStart } from 'kibana/public';
+import { matchPath } from 'react-router-dom';
+import { AppLocation, Immutable } from '../../../../../common/endpoint/types';
+import { ImmutableMiddleware, ImmutableMiddlewareAPI } from '../../../../common/store';
+import { AppAction } from '../../../../common/store/actions';
+import { MANAGEMENT_ROUTING_HOST_ISOLATION_EXCEPTIONS_PATH } from '../../../common/constants';
+import { parseQueryFilterToKQL } from '../../../common/utils';
+import {
+  createFailedResourceState,
+  createLoadedResourceState,
+} from '../../../state/async_resource_builders';
+import { getHostIsolationExceptionsList } from '../service';
+import { HostIsolationExceptionsPageState } from '../types';
+import { getCurrentListPageDataState, getCurrentLocation } from './selector';
+
+export const SEARCHABLE_FIELDS: Readonly<string[]> = [`name`, `description`, `entries.value`];
+
+export function hostIsolationExceptionsMiddlewareFactory(coreStart: CoreStart) {
+  return createHostIsolationExceptionsPageMiddleware(coreStart);
+}
+
+export const createHostIsolationExceptionsPageMiddleware = (
+  coreStart: CoreStart
+): ImmutableMiddleware<HostIsolationExceptionsPageState, AppAction> => {
+  return (store) => (next) => async (action) => {
+    next(action);
+
+    if (action.type === 'userChangedUrl' && isHostIsolationExceptionsPage(action.payload)) {
+      loadHostIsolationExceptionsList(store, coreStart.http);
+    }
+  };
+};
+
+async function loadHostIsolationExceptionsList(
+  store: ImmutableMiddlewareAPI<HostIsolationExceptionsPageState, AppAction>,
+  http: HttpStart
+) {
+  const { dispatch } = store;
+  try {
+    const {
+      page_size: pageSize,
+      page_index: pageIndex,
+      filter,
+    } = getCurrentLocation(store.getState());
+    const query = {
+      http,
+      page: pageIndex + 1,
```

Comment:
can you change the default index value to 1 so you don't need to increment it here?

Reply:
I thought of it but I went against it. Somehow the default pageIndex for the management page is 0, and that's what I am using as the initial value.